pkg/crypto/aes.go

```diff
@@ -81,35 +89,27 @@ func Encrypt(passphrase string, plaintext []byte, encoding string) (string, erro
 
 	data := aesgcm.Seal(nil, iv, plaintext, nil)
 
-	var prefix string
 	var encodedSalt string
 	var encodedIV string
 	var encodedData string
 	if encoding == "base64" {
-		prefix = Base64EncodingPrefix
 		encodedSalt = base64.StdEncoding.EncodeToString(salt)
 		encodedIV = base64.StdEncoding.EncodeToString(iv)
 		encodedData = base64.StdEncoding.EncodeToString(data)
 	} else if encoding == "hex" {
-		prefix = HexEncodingPrefix
 		encodedSalt = hex.EncodeToString(salt)
 		encodedIV = hex.EncodeToString(iv)
 		encodedData = hex.EncodeToString(data)
 	} else {
 		return "", errors.New("Invalid encoding, must be one of [base64, hex]")
 	}
```

Comment:
It seems to be more idiomatic and recommended to use switch of if else if else statements:
> Switch
> Go's switch is more general than C's. The expressions need not be constants or even integers, the cases are evaluated top to bottom until a match is found, and if the switch has no expression it switches on true. It's therefore possible—and idiomatic—to write an if-else-if-else chain as a switch.

nit: Convert if else if else chain to tagged switch:
https://staticcheck.io/docs/checks#QF1003 (linter)

```go
switch encoding {
case "base64":
	encodedSalt = base64.StdEncoding.EncodeToString(salt)
	encodedIV = base64.StdEncoding.EncodeToString(iv)
	encodedData = base64.StdEncoding.EncodeToString(data)
case "hex":
	encodedSalt = hex.EncodeToString(salt)
	encodedIV = hex.EncodeToString(iv)
	encodedData = hex.EncodeToString(data)
default:
	return "", errors.New("Invalid encoding, must be one of [base64, hex]")
}
```

Reply:
Same thing for this portion:

```go
switch rawEncoding {
case Base64EncodingPrefix:
	encoding = Base64EncodingPrefix
case hexEncodingPrefix, "":
	// default to hex for backwards compatibility b/c we didn't always include an encoding prefix
	// TODO remove support for optional prefix when releasing CLI v4 (DPLR-435)
	encoding = hexEncodingPrefix
default:
	return "", errors.New("Invalid encoding, must be one of [base64, hex]")
}
```

Reply:
This seems more like personal preference rather than correctness. I'm going to stick with the current approach rather than refactor all of our conditionals since it doesn't seem we gain anything. If we want to make this codebase more idiomatic, I'd push for investing our time in adding support for a `context`.